src/vsr/replica.zig

```diff
@@ -2285,34 +2277,11 @@ pub fn ReplicaType(
                     command,
                     message.header.replica,
                 });
-                return null;
-            }
-
-            // Record the first receipt of this message:
-            assert(messages[message.header.replica] == null);
-            messages[message.header.replica] = message.ref();
-
-            // Count the number of unique messages now received:
-            const count = self.count_quorum(messages, message.header.command, message.header.context);
-            log.debug("{}: on_{s}: {} message(s)", .{ self.replica, command, count });
-
-            // Wait until we have exactly `threshold` messages for quorum:
-            if (count < threshold) {
-                log.debug("{}: on_{s}: waiting for quorum", .{ self.replica, command });
-                return null;
-            }
-
-            // This is not the first time we have had quorum, the state transition has already happened:
-            if (count > threshold) {
-                log.debug("{}: on_{s}: ignoring (quorum received already)", .{
-                    self.replica,
-                    command,
-                });
-                return null;
+            } else {
```

Comment:
`// Do not allow duplicate messages to trigger multiple passes through a state transition:` commend no longer makes sense, as we now return `void`